app/main/views/services.py

```diff
@@ -102,24 +101,36 @@ def update_service_status(service_id):
 @login_required
 @flask_featureflags.is_active_feature('EDIT_SERVICE_PAGE')
 def edit_section(service_id, section):
-
-    service = data_api_client.get_service(service_id)['services']
+    service = data_api_client.get_service(service_id)
+    if service is None:
+        abort(404)
+    service = service['services']
 
     if not _is_service_associated_with_supplier(service):
         abort(404)
 
     content = existing_service_content.get_builder().filter(service)
+    section = content.get_section(section)
+    if section is None:
+        abort(404)
 
     return render_template(
         "services/edit_section.html",
-        section=content.get_section(section),
+        section=section,
+        page_questions='|'.join(get_section_questions(section)),
```

Comment:
I'm having trouble figuring out what `page_questions` is for. It's set as a hidden input field whenever the template is rendered (even though we seem to have access to `section` which we can get the questions from), but then https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/105/files#diff-32bf67afda65ad26faef3bf3efd522b1R155 just removes it right away without using it. Am I missing something?